### ISSUE TITLE
TagsToShortcodeTask breaking content

### ISSUE THREAD

**Issue by ichaber** [silverstripe/silverstripe-assets#320]
When running TagsToShortcodeTask on SilverStripe 4.4.0 and assets in version 1.4.3 sometimes the Content is replaced with an invalid markup.

E.g.: The following code in a SilverStripe 3 format
```
<p><a><img class="leftAlone" title="" src="assets/media-releases/img/_resampled/ResizedImageAmNsTLr4AXNQ/20190101-harris-lane-6.jpg" alt="" width="600" height="400"></a></p>
```
will be rewritten to the following markup:
```
<p><a id="2"][image class="leftAlone" src="/assets/44ad6c19ab/image.jpg" width="600" height="400" id="2"]</a></p>
```

# Pull Request
* [ ] https://github.com/silverstripe/silverstripe-assets/pull/321

**Comment by maxime-rainville** [comment]
@bergice I've assigned you for peer review as you wrote the original task and you will have the most context. Let me know if this is problematic in any way.

**Comment by andreaslindahl** [comment]
I'm experiencing this bug when migrating from Silverstripe 3 to 4.5. Has this been fixed?

**Comment by Cheddam** [comment]
@andreaslindahl The [bugfix](https://github.com/silverstripe/silverstripe-assets/commit/37d8923503678e41b277bdfc9537dbaeb8f5ae79) has been included in CMS 4.4.4-4.5.0 so far, so you shouldn't be experiencing issues. Could you provide some more context (which version of CMS 3 you're migrating from, any broken content samples, etc.)?
